### PR TITLE
Quest profiles - resolve some conflicts, fix upgrade crash

### PR DESCRIPTION
**Pull request by matkoniecz**
Note: `streetcomplete:quest-profiles` not `streetcomplete:master` is the merge target.

Updates quest profiles branch (later than quest-order and containing it) to https://github.com/streetcomplete/StreetComplete/commit/94329195cdf6a5be2198bb630bbf9b334bf14a23 - still without my recent changes (I want to confirm that I am now doing it in a proper way before reapplying them)

Resolves some conflicts along way in QuestModule and changelog.yml (because new quest AddCrossing got added)

BTW, it seems that quest-order pointer can be safely deleted.

---------------

This was tested. It includes also a fix for an upgrade crash

Crash can be reproduced by 

- checkouting a69abe21b0d51b3f327897b636f4b13f8613d0c0
- running app in emulator - so next run will be upgrading app where bug will be triggered
- checkouting df3dd4c83
- running app in emulator

Fix https://github.com/streetcomplete/StreetComplete/pull/3193/commits/1b74915eb445b8aa787d4114a67bcaecfc01db9f can be reproduced by
- checkouting a69abe21b0d51b3f327897b636f4b13f8613d0c0
- running app in emulator
- checkouting 1b74915eb
- running app in emulator

it produced

```
E/SQLiteLog: (1) near ")": syntax error
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.westnordost.streetcomplete.debug, PID: 6241
    java.lang.RuntimeException: Unable to start activity ComponentInfo{de.westnordost.streetcomplete.debug/de.westnordost.streetcomplete.MainActivity}: android.view.InflateException: Binary XML file line #15 in de.westnordost.streetcomplete.debug:layout/activity_main: Binary XML file line #15 in de.westnordost.streetcomplete.debug:layout/activity_main: Error inflating class fragment
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3270)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: android.view.InflateException: Binary XML file line #15 in de.westnordost.streetcomplete.debug:layout/activity_main: Binary XML file line #15 in de.westnordost.streetcomplete.debug:layout/activity_main: Error inflating class fragment
     Caused by: android.view.InflateException: Binary XML file line #15 in de.westnordost.streetcomplete.debug:layout/activity_main: Error inflating class fragment
     Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment de.westnordost.streetcomplete.map.MainFragment: calling Fragment constructor caused an exception
        at androidx.fragment.app.Fragment.instantiate(Fragment.java:631)
        at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
        at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:483)
        at androidx.fragment.app.FragmentLayoutInflaterFactory.onCreateView(FragmentLayoutInflaterFactory.java:94)
        at androidx.fragment.app.FragmentController.onCreateView(FragmentController.java:135)
        at androidx.fragment.app.FragmentActivity.dispatchFragmentsOnCreateView(FragmentActivity.java:319)
        at androidx.fragment.app.FragmentActivity.onCreateView(FragmentActivity.java:298)
        at android.view.LayoutInflater.tryCreateView(LayoutInflater.java:1069)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:997)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:1123)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1084)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:682)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:534)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:481)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:699)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:195)
        at de.westnordost.streetcomplete.MainActivity.onCreate(MainActivity.kt:100)
        at android.app.Activity.performCreate(Activity.java:7802)
        at android.app.Activity.performCreate(Activity.java:7791)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1299)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
E/AndroidRuntime:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at androidx.fragment.app.Fragment.instantiate(Fragment.java:613)
        	... 32 more
     Caused by: android.database.sqlite.SQLiteException: near ")": syntax error (code 1 SQLITE_ERROR): , while compiling: INSERT INTO quest_visibility (
        quest_preset_id,
        quest_type,
        visibility,
    ) SELECT
        0,
        quest_type,
        visibility
    FROM quest_visibility_old;
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:986)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:593)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:61)
        at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:33)
        at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1805)
        at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1733)
        at de.westnordost.streetcomplete.data.StreetCompleteSQLiteOpenHelper.onUpgrade(StreetCompleteSQLiteOpenHelper.kt:113)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:417)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:317)
        at de.westnordost.streetcomplete.data.AndroidDatabase.getDb(AndroidDatabase.kt:15)
        at de.westnordost.streetcomplete.data.AndroidDatabase.query(AndroidDatabase.kt:57)
        at de.westnordost.streetcomplete.data.Database$DefaultImpls.query$default(Database.kt:24)
        at de.westnordost.streetcomplete.data.osm.edits.ElementEditsDao.getAllUnsynced(ElementEditsDao.kt:78)
        at de.westnordost.streetcomplete.data.osm.edits.ElementEditsController.getAllUnsynced(ElementEditsController.kt:60)
        at de.westnordost.streetcomplete.data.osm.edits.MapDataWithEditsSource.rebuildLocalChanges(MapDataWithEditsSource.kt:346)
        at de.westnordost.streetcomplete.data.osm.edits.MapDataWithEditsSource.<init>(MapDataWithEditsSource.kt:145)
        at de.westnordost.streetcomplete.data.osm.edits.MapDataWithEditsSource_Factory.newInstance(MapDataWithEditsSource_Factory.java:45)
        at de.westnordost.streetcomplete.data.osm.edits.MapDataWithEditsSource_Factory.get(MapDataWithEditsSource_Factory.java:32)
        at de.westnordost.streetcomplete.data.osm.edits.MapDataWithEditsSource_Factory.get(MapDataWithEditsSource_Factory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at de.westnordost.streetcomplete.data.osm.osmquests.OsmQuestController_Factory.get(OsmQuestController_Factory.java:47)
        at de.westnordost.streetcomplete.data.osm.osmquests.OsmQuestController_Factory.get(OsmQuestController_Factory.java:13)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at de.westnordost.streetcomplete.data.quest.QuestController_Factory.get(QuestController_Factory.java:43)
        at de.westnordost.streetcomplete.data.quest.QuestController_Factory.get(QuestController_Factory.java:13)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at de.westnordost.streetcomplete.DaggerApplicationComponent.injectMainFragment(DaggerApplicationComponent.java:1070)
        at de.westnordost.streetcomplete.DaggerApplicationComponent.inject(DaggerApplicationComponent.java:831)
        at de.westnordost.streetcomplete.map.MainFragment.<init>(MainFragment.kt:137)
        	... 35 more
```

crash was on 

```
                    INSERT INTO quest_visibility (
                        quest_preset_id,
                        quest_type,
                        visibility
                    ) SELECT
                        0,
                        quest_type,
                        visibility
                    FROM quest_visibility_old;
```

query

Insert here rant about any system not allowing trailing commas or demanding to not include them. It results in uglier diffs and completely pointless bugs like this one. For no benefit at all.

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/update_tags/StringMapChangesBuilder.kt [hunk 1]
```diff
@@ -24,15 +24,15 @@ class StringMapChangesBuilder(private val source: Map<String, String>) {
     }
 
     fun add(key: String, value: String) {
-        require(!source.containsKey(key)) { "The key '$key' already exists in the map." }
+        require(!source.containsKey(key)) { "The key '$key' already exists in the map '$source." }
```
Comment (peternewman) [inline app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/update_tags/StringMapChangesBuilder.kt:27]: Add a missing quote:
```suggestion
        require(!source.containsKey(key)) { "The key '$key' already exists in the map '$source'." }
```

**Review by peternewman — CHANGES_REQUESTED** [review]
One minor typo.

This would be a lot less confusing to review if there was a separate PR updating the quest-profiles branch, and then this would just show the direct changes. I don't even know if the typo I found is your change or not.

**Review by westnordost — APPROVED** [review]

**Comment by westnordost** [comment]
ok, thanks

File: app/src/main/java/de/westnordost/streetcomplete/data/StreetCompleteSQLiteOpenHelper.kt [hunk 2]
```diff
@@ -114,7 +114,7 @@ import de.westnordost.streetcomplete.quests.oneway_suspects.data.WayTrafficFlowT
                 INSERT INTO ${VisibleQuestTypeTable.NAME} (
                     ${VisibleQuestTypeTable.Columns.QUEST_PRESET_ID},
                     ${VisibleQuestTypeTable.Columns.QUEST_TYPE},
-                    ${VisibleQuestTypeTable.Columns.VISIBILITY},
+                    ${VisibleQuestTypeTable.Columns.VISIBILITY}
                 ) SELECT
```
Comment (smichel17) [inline app/src/main/java/de/westnordost/streetcomplete/data/StreetCompleteSQLiteOpenHelper.kt:118]: > Insert here rant about any system not allowing trailing commas or demanding to not include them. It results in uglier diffs and completely pointless bugs like this one. For no benefit at all.

Note that you can at least improve the situation using leading comma formatting, although it seems to mostly/only be popular in functional programming circles.

```
                INSERT INTO ${VisibleQuestTypeTable.NAME}
                    ( ${VisibleQuestTypeTable.Columns.QUEST_PRESET_ID}
                    , ${VisibleQuestTypeTable.Columns.QUEST_TYPE}
                    , ${VisibleQuestTypeTable.Columns.VISIBILITY}
                    )
                SELECT
```
Admittedly it doesn't work outside of parentheses, like in the SELECT below.

Comment (peternewman) [reply app/src/main/java/de/westnordost/streetcomplete/data/StreetCompleteSQLiteOpenHelper.kt:118]: Given it's just building a query, and they're all constants so the raw query is useless on it's own, what about joining a set with commas, which would be safe from all of that and self-validating via the compiler?

**Pull request merged**